scio-test/src/main/scala/com/spotify/scio/testing/TransformOverride.scala

```diff
@@ -105,6 +98,43 @@ object TransformOverride {
     PTransformOverride.of(new EqualNamePTransformMatcher(name), overrideFactory)
   }
 
+  /**
+   * @return
+   *   A [[PTransformOverride]] which when applied will override a [[PTransform]] with name `name`
+   *   with a transform flat-mapping elements via `fn`.
+   */
+  def off[T: ClassTag, U](name: String, fn: T => Iterable[U]): PTransformOverride = {
+    val wrappedFn: T => lang.Iterable[U] = fn.compose { t: T =>
+      typeValidation(
+        s"Input for override transform $name does not match pipeline transform.",
+        t.getClass,
+        implicitly[ClassTag[T]].runtimeClass
+      )
+      t
+    }.andThen(_.asJava)
+
+    val overrideFactory =
+      factory[PCollection[T], PCollection[U], PTransform[PCollection[T], PCollection[U]]](
+        t => PTransformReplacements.getSingletonMainInput(t),
+        new PTransform[PCollection[T], PCollection[U]]() {
+          override def expand(input: PCollection[T]): PCollection[U] = {
+            val sf: SimpleFunction[T, lang.Iterable[U]] = Functions.simpleFn(wrappedFn)
+            input.apply(FlatMapElements.via(sf))
```

Comment:
this line causes `interface java.lang.Iterable is not a super class of U`. IIUC  the correct type is passed:

`  public static <InputT, OutputT> FlatMapElements<InputT, OutputT> via(`
 `SimpleFunction<? super InputT, ? extends Iterable<OutputT>> fn) {` 

And I make sure I pass a java iterable. Any ideas?

Reply:
Update: this one is fixed by creating an InferableFunction directly, instead of using `Functions.simpleFn`

Reply:
ah, i see! Maybe worth adding `InferableFunction` support to our `Functions` class :)